ocaml/xapi/xapi_pool.ml

```diff
@@ -1745,29 +1745,40 @@ let eject_self ~__context ~host =
       )
       my_vms_with_records ;
     (* all control domains resident on me should be destroyed once I leave the
-       		pool, therefore pick them out as follows: if they have a valid resident_on,
-       		the latter should be me; if they don't (e.g. they are halted), they should have
-       		disks on my local storage *)
+       pool, therefore pick them out as follows: if they have a valid
+       resident_on, the latter should be me; if they don't (e.g. they are
+       halted), they should have disks on my local storage *)
     let vm_is_resident_on_host vm_rec host =
       Db.is_valid_ref __context vm_rec.API.vM_resident_on
       && vm_rec.API.vM_resident_on = host
     in
-    let vm_has_disks_on_local_sr_of_host vm_ref host =
-      let is_sr_local x = not (Helpers.is_sr_shared ~__context ~self:x) in
+    let vm_has_disks_on_local_sr_of_host vm_rec host =
+      let is_sr_local x =
+        try not (Helpers.is_sr_shared ~__context ~self:x) with _ -> false
```

Comment:
Should we catch just the specific error for invalid reference here instead of a catch all?
E.g. if we have some other genuine error we should probably surface that.